### PR TITLE
add, delete, check, delete all, autosave, blur,mui

### PR DESCRIPTION
**Pull request by ikirienko-exceedteam**

### CODE REVIEW

File: src/components/TodoList/TodoList.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
+/* eslint-disable no-unused-vars */
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import Button from "@material-ui/core/Button";
```
Comment (usatovaDasha) [inline src/components/TodoList/TodoList.js:4]: core in one import

File: src/components/TodoList/TodoList.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import "./TodoList.scss";
+
+const TodoList = () => {
+  const [tasks, setTasks] = useState([]);
+  const [inputText, setInputText] = useState("");
+  const [editIndex, setEditIndex] = useState(-1);
+  const [editText, setEditText] = useState("");
+
+  let flagCheckBoxEditSaveDelCancelProcessed = false;
```
Comment (usatovaDasha) [inline src/components/TodoList/TodoList.js:21]: don't use global variables. Using state

File: src/components/TodoList/TodoList.js [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+  const onEnterDownAddTaskInput = (event) => {
+    if (event.code === "Enter") {
+      addNewTask();
+    }
+  };
+
+  const addNewTask = async () => {
+    if (isEmpty(inputText)) {
+      alert("First input the text of the task");
```
Comment (usatovaDasha) [inline src/components/TodoList/TodoList.js:54]: don't use alert. using snackbar material

File: src/components/TodoList/TodoList.js [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+    setEditText(tasks[index].text);
+    flagCheckBoxEditSaveDelCancelProcessed = true;
+    setEditIndex(-1);
+  };
+
+  const onChangeCheckbox = async (e, task) => {
+    flagCheckBoxEditSaveDelCancelProcessed = true;
+
+    const body = {
+      _id: task._id,
```
Comment (usatovaDasha) [inline src/components/TodoList/TodoList.js:161]: destructurization

File: src/components/TodoList/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+        <TextField
+          label="Your task"
+          variant="outlined"
+          className="add-task-text"
+          autoFocus
+          value={inputText}
+          onChange={(e) => setInputText(e.target.value)}
+          onKeyDown={onEnterDownAddTaskInput}
+        />
+        <Button variant="contained" size="large" color="primary" onClick={addNewTask}>
```
Comment (usatovaDasha) [inline src/components/TodoList/TodoList.js:215]: move to single lines

File: src/components/TodoList/TodoList.scss [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+  .title {
+    font-size: calc(30px + 3vw);
+    font-family: $fontFamilyTitle;
+    margin: 30px auto;
+    color: #f50057;
+    font-weight: 800;
+  }
+
+  .main-elements-control {
+    // width: 100%;
```
Comment (usatovaDasha) [inline src/components/TodoList/TodoList.scss:29]: remove comments

File: src/components/TodoList/TodoList.js (hunk 2 above)
Comment (ikirienko-exceedteam) [reply src/components/TodoList/TodoList.js:28]: Variable is intentionally used here. Unfortunately any state worked in this certain situation.

**Pull request merged**